content/en/docs/concepts/extend-kubernetes/api-extension/custom-resources.md

```diff
@@ -9,40 +9,49 @@ weight: 20
-Kubernetes cluster.
-In other words, it represents a customization of a particular Kubernetes installation.
+A *custom resource* is an extension of the Kubernetes API that is not necessarily available in a default
+Kubernetes installation. It represents a customization of a particular Kubernetes installation. However,
+many core Kubernetes functions are now built using custom resources, making Kubernetes more modular.
 
 Custom resources can appear and disappear in a running cluster through dynamic registration,
 and cluster admins can update custom resources independently of the cluster itself.
-Once a custom resource is installed, users can create and access its objects with
-[kubectl](/docs/user-guide/kubectl-overview/), just as they do for built-in resources like *pods*.
+Once a custom resource is installed, users can create and access its objects using the
+[kubectl](/docs/user-guide/kubectl-overview/) command, just as they do for built-in resources like
+*Pods*.
 
-### Custom controllers
+## Custom controllers
 
 On their own, custom resources simply let you store and retrieve structured data.
-It is only when combined with a *controller* that they become a true declarative API.
+When you combine a custom resource with a *custom controller*, custom resources
+provide a true _declarative API_.
+
 A [declarative API](/docs/concepts/overview/working-with-objects/kubernetes-objects/#understanding-kubernetes-objects)
 allows you to _declare_ or specify the desired state of your resource and tries
-to match the actual state to this desired state.
-Here, the controller interprets the structured data as a record of the user's
-desired state, and continually takes action to achieve and maintain this state.
+keep the current state of Kubernetes objects in sync with the desired state.
+The controller interprets the structured data as a record of the user's
+desired state, and continually maintains this state (or generates errors if it
+cannot maintain the state).
 
-A *custom controller* is a controller that users can deploy and update on a running cluster, independently of the cluster's own lifecycle. Custom controllers can work with any kind of resource, but they are especially effective when combined with custom resources. The [Operator](https://coreos.com/blog/introducing-operators.html) pattern is one example of such a combination. It allows developers to encode domain knowledge for specific applications into an extension of the Kubernetes API.
+You can deploy and update a custom controller on a running cluster, independently
+of the cluster's own lifecycle. Custom controllers can work with any kind of resource,
+but they are especially effective when combined with custom resources. The
+[Operator pattern](https://coreos.com/blog/introducing-operators.html) combines a custom
+resource and a custom controller. You can use custom controllers to encode domain knowledge
```

Comment:
Maybe "Operator pattern..combines custom resources and custom controllers" instead of a **_a_** custom resource and **_a_** custom controller? The idea is to make sure that it is not necessary that there is only one custom resource or only one custom controller, but I'm not sure what is the right way to write it. :grimacing: